src/main_context_channel.rs

```diff
@@ -363,9 +369,15 @@ impl<T> Drop for Sender<T> {
 /// See [`MainContext::sync_channel()`] for how to create such a `SyncSender`.
 ///
 /// [`MainContext::sync_channel()`]: struct.MainContext.html#method.sync_channel
-#[derive(Clone, Debug)]
+#[derive(Debug)]
```

Comment:
Shouldn't we do the same for `Debug`?

Reply:
No, we can only have a `Debug` impl if `T: Debug`. It is printing the queued items.

Reply:
That's what I said: isn't it forcing `T` to implement `Debug`?

Reply:
Yes, so that it can print the currently queued up items.

Reply:
For context, the `std::mpsc` `Sender` / `Receiver` have a manual `Debug` impl that work with any type.

I guess for consistency we should do the same. Means that `Debug` is available always, and that we print fewer details.

What do you think?

Reply:
I'd prefer consistency btw

Reply:
I prefer consistency too.

Reply:
Done